packages/core/core/src/ParcelConfig.js

```diff
@@ -332,6 +422,47 @@ export default class ParcelConfig {
     return this.loadPlugins<Reporter>(this.reporters);
   }
 
+  resolveTransformers(
+    filePath: FilePath,
+    pipeline?: ?string,
+    allowEmpty?: boolean,
+  ): Promise<
+    Array<{|
+      name: string,
+      resolved: FilePath,
+      resolveFrom: FilePath,
+      version: Semver,
+    |}>,
+  > {
+    return this.resolvePlugins(
+      this._getTransformerNodes(filePath, pipeline, allowEmpty),
+    );
+  }
+
+  resolveValidators(
+    filePath: FilePath,
+  ): Promise<
+    Array<{|
+      name: string,
+      resolved: FilePath,
+      resolveFrom: FilePath,
+      version: Semver,
+    |}>,
+  > {
+    return this.resolvePlugins(this._getValidatorNodes(filePath));
+  }
+
+  resolveResolvers(): Promise<
+    Array<{|
+      name: string,
+      resolved: FilePath,
+      resolveFrom: FilePath,
+      version: Semver,
+    |}>,
+  > {
+    return this.resolvePlugins(this._getResolverNodes());
+  }
+
```

Comment:
Could you add `resolveBundler`, `resolveRuntimes`, `resolvePackager`, and `resolveOptimizers` as well? 🙏

Reply:
For these, should they also be called in `ConfigLoader.js`? Are there `actionTypes` for bundler, runtimes, packager, optimizers? 

https://github.com/parcel-bundler/parcel/blob/676499e3565d531e86178d978f89375d86fddf28/packages/core/core/src/ConfigLoader.js#L61-L75